test/e2e/discovery/service_discovery.go

```diff
@@ -81,6 +81,21 @@ var _ = Describe("[discovery] Test Service Discovery Across Clusters", func() {
 			RunServiceDiscoveryRoundRobinTest(f)
 		})
 	})
+
+	When("only one of the clusters with a service is healthy", func() {
+		var healthCheckIP, endpointName string
+		BeforeEach(func() {
+			randomIP := "192.168.1.5"
+			endpointName, healthCheckIP = f.GetHealthCheckIPInfo(framework.ClusterC)
+			f.SetHealthCheckIP(framework.ClusterC, randomIP, endpointName)
+		})
+		It("should not resolve the service", func() {
+			RunServicesClusterAvailabilityMutliClusterTest(f)
+		})
+		AfterEach(func() {
+			f.SetHealthCheckIP(framework.ClusterC, healthCheckIP, endpointName)
+		})
```

Comment:
Can you put newlines in between the functions (I'm surprised the linter didn't complain)? Scrunched code is less readable.